plugins/GrapesJsBuilderBundle/EventSubscriber/EmailSubscriber.php

```diff
@@ -54,4 +81,35 @@ public function onEmailDelete(Events\EmailEvent $event): void
             $this->grapesJsBuilderModel->getRepository()->deleteEntity($grapesJsBuilder);
         }
     }
+
+    public function manageEmailDraft(Events\EmailEditSubmitEvent $event): void
+    {
+        if (!$this->config->isPublished()) {
+            return;
+        }
+
+        $email           = $event->getCurrentEmail();
+        $grapesJsBuilder = $this->grapesJsBuilderModel->getRepository()->findOneBy(['email' => $email]);
+
+        if ($event->isSaveAsDraft()) {
+            // Set draft MJML and restore previous version when saving a draft
+            $grapesJsBuilder->setDraftCustomMjml($grapesJsBuilder->getCustomMjml());
+            $grapesJsBuilder->setCustomMjml($this->existingMjml);
+
+            // reset the html of the parent email as well
+            $email->setCustomHtml($this->existingHtml);
```

Comment:
There is no need to update `CustomHtml` as we are saving only draft and not changing anything in original email content.

Reply:
When saving as draft the customHtml of the email entity is updated, this line resets it back to what it was.